File: algorithms/[674]最长连续递增序列/solution.py
# topics = ["数组"]

from typing import List


class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        n = len(nums)

        if n == 0 or n == 1:
            return n

        res: List[int] = [1]
        for idx in range(1, n):
            if nums[idx] > nums[idx - 1]:
                res[-1] += 1
            else:
                res.append(1)

        return max(res)

File: algorithms/[674]最长连续递增序列/solution.ts
// topics = ["数组"]

function findLengthOfLCIS(nums: number[]): number {
  const len = nums.length;

  if (len === 0 || len === 1) {
    return len;
  }

  let res = 0;
  // 记录当前连续递增序列起始索引
  let start = 0;
  for (let i = 1; i < len; i += 1) {
    if (nums[i] <= nums[i - 1]) {
      res = Math.max(res, i - start);
      start = i;
    }
  }

  return Math.max(res, len - start);
}
